Owncloud iOs Client/Login/Login/UniversalLoginViewController.swift

```diff
@@ -538,53 +537,54 @@ public enum TextfieldType: String {
             self.setNetworkActivityIndicator(status: true)
             // get public infor from server
             getPublicInfoFromServerJob.start(serverURL: self.serverURLNormalizer.normalizedURL, withCompletion: { (validatedURL: String?, _ serverAuthenticationMethods: Array<Any>?, _ error: Error?, _ httpStatusCode: NSInteger) in
-            
-                self.setNetworkActivityIndicator(status: false)
-                if (error != nil || validatedURL == nil) {
-                    
-                    self.setConnectButton(status: false)
-                    self.showURLError(
-                        self.manageNetworkErrors.returnErrorMessage(
-                            withHttpStatusCode: httpStatusCode, andError: error
+                
+                DispatchQueue.main.async {
+                    self.setNetworkActivityIndicator(status: false)
+                    if (error != nil || validatedURL == nil) {
+                        
+                        self.setConnectButton(status: false)
+                        self.showURLError(
+                            self.manageNetworkErrors.returnErrorMessage(
+                                withHttpStatusCode: httpStatusCode, andError: error
+                            )
                         )
-                    )
-                    print ("error getting information from URL")
-                    
-                } else if validatedURL != nil {
-                    
-                    self.setURLFooter(isType: .None)
-                    
-                    self.validatedServerURL = validatedURL;
-                    self.allAvailableAuthMethods = serverAuthenticationMethods as! [AuthenticationMethod]
-                    
-                    self.authMethodToLogin = DetectAuthenticationMethod.getAuthMethodToLoginFrom(availableAuthMethods: self.allAvailableAuthMethods)
-                    
-                    if (self.authMethodToLogin != .NONE) {
-                        self.setReconnectionButtons(hiddenStatus: true)
-
-                        if (self.authMethodToLogin == .BASIC_HTTP_AUTH) {
-                            self.textFieldURL.resignFirstResponder()
-                            self.textFieldUsername.becomeFirstResponder()
+                        print ("error getting information from URL")
+                        
+                    } else if validatedURL != nil {
+                        
+                        self.setURLFooter(isType: .None)
+                        
+                        self.validatedServerURL = validatedURL;
```

Comment:
I think here you are using three global variables. I think we should avoid the use of global variables until it's completely needed. The use of global variables when they are not needed may lead into unexpected behaviour and tedious debugging in the future.

Reply:
Totally agree, this was not added in this pr. We can consider refactor in future.